common/trace.go

```diff
@@ -0,0 +1,58 @@
+package common
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+
+	browsertrace "github.com/grafana/xk6-browser/trace"
+)
+
+// Tracer defines the interface with the tracing methods used in the common package.
+type Tracer interface {
+	TraceAPICall(
+		ctx context.Context, targetID string, spanName string, opts ...trace.SpanStartOption,
+	) (context.Context, trace.Span)
+	TraceNavigation(
+		ctx context.Context, targetID string, url string, opts ...trace.SpanStartOption,
+	) (context.Context, trace.Span)
+	TraceEvent(
+		ctx context.Context, targetID string, eventName string, spanID string, opts ...trace.SpanStartOption,
+	) (context.Context, trace.Span)
+}
+
+// TraceAPICall is a helper method that retrieves the Tracer from the given ctx and
+// calls its TraceAPICall implementation. If the Tracer is not present in the given
+// ctx, it returns a noopSpan and the given context.
+func TraceAPICall(
+	ctx context.Context, targetID string, spanName string, opts ...trace.SpanStartOption,
+) (context.Context, trace.Span) {
+	if tracer := GetTracer(ctx); tracer != nil {
+		return tracer.TraceAPICall(ctx, targetID, spanName, opts...)
+	}
```

Comment:
I considered panicking here, as the `tracer` should always be present in the context, whether it is a _noop_ or a "valid implementation", but that would require us to mock it in all of our tests. Also, we already have a test to verify that the tracer is correctly created from the `VU.State.TracerProvider` and propagated to all the components.

Reply:
I wonder if it's worth adding a warning log when we do return a `NoopSpan`? Is there a valid case for tracer to ever be `nil`?

Reply:
Reviewing this, adding a warning log when the tracer is not present in the context should be easy, but we'll have to accept the logger as input for the common package tracing functions. Take into account also that we have an integration test that already verifies that the tracer is present and propagated to all components that currently generate traces.